pallets/mixer/src/lib.rs

```diff
@@ -145,15 +143,13 @@ pub mod pallet {
 	#[pallet::metadata(<T as frame_system::Config>::AccountId = "AccountId", <T as merkle::Config>::GroupId = "GroupId")]
 	pub enum Event<T: Config> {
 		/// New deposit added to the specific mixer
-		Deposit(
-			<T as merkle::Config>::GroupId,
-			<T as frame_system::Config>::AccountId,
-			ScalarData,
-		),
+		Deposit(<T as merkle::Config>::GroupId, <T as frame_system::Config>::AccountId),
```

Comment:
Why here we removed the `ScalarData` or the leaf that has been deposited?

Reply:
I think we should add the leaf an the index of the deposit to the event so that:
- Anyone consuming events can rebuild the tree without anything else besides Deposit events
- The user depositing knows on successful deposit what index their deposit is for (to add to the note).

Reply:
Maybe its best to do this in merkle pallet, with event `NewMembers(id, sender, leaf_count_before, leaves_array)` where `leaf_count_before` is number of leaves before insertions, and `leaves_array` is the passed array of leaves

Reply:
I think we should still add more to this event. Namely, we should be adding the amount of the deposit and the index. Lets just replicate info on events since its easy to do at the moment and will be useful for developing against it.

Reply:
Ok, adding amount, number of leaves before insertion, and an array of leaves inside the deposit event